addon/helpers/object.js

```diff
@@ -4,6 +4,10 @@ import { reduceObject } from '../utils/object-utils';
 
 export default MicroState.extend({
 
+  initialValueFor([object = {}]) {
+    return object;
```

Comment:
All this thinking about coercion has got me wondering about massaging values coming into the Object constructor. We could try to massage it, which seems complex, or we could throw an exception if the value passed in is primitive.

```js
if (isPrimitive(object)) {
  throw new Error(`Object microstate expects a non primitive value in its constructor, but received ${object}`);
}

function isPrimitive(value) {
  let type = typeof value;
  return type != "object" && type != "function"
}
```